packages/hydrogen/src/seo/seo.ts

```diff
@@ -39,59 +40,104 @@ export function Seo({debug}: SeoProps) {
   const matches = useMatches();
   const location = useLocation();
 
-  const seoConfig = matches
-    .flatMap((match) => {
-      const {handle, ...routeMatch} = match;
-      const routeInfo = {...routeMatch, ...location};
-
-      if (handle === undefined || handle.seo === undefined) {
-        if (!routeMatch?.data?.seo) {
-          return [];
-        }
-
-        return [
-          {
-            ...routeMatch.data.seo,
-          },
-        ];
+  // Capture the seo and jsonLd configs from the route matches
+  const seoConfig = useMemo(() => {
+    return (
+      matches
+        .flatMap((match) => {
+          const {handle, ...routeMatch} = match;
+          const routeData = {...routeMatch, ...location};
+          const handleSeo = handle?.seo;
+          const loaderSeo = routeMatch?.data?.seo;
+
+          if (!handleSeo && !loaderSeo) {
+            return [];
+          }
+
+          // if seo is defined in the handle, invoke it with the route data
+          if (handleSeo) {
+            return recursivelyInvokeOrReturn(handle.seo, routeData);
+          } else {
+            // let jsonLd = loaderSeo?.jsonLd;
+            // if (jsonLd) {
+            //   jsonLd = {...jsonLd, routeId: routeData.id};
+            // }
+            // return [{...loaderSeo, jsonLd}];
+            return [loaderSeo];
+          }
+        })
+        // merge route seo (priority) with the root seo if both are present
+        // jsonLd definitions are instead concatenated because there can be
+        // multiple jsonLd tags on any given root+route. e.g root renders Organization
+        // schema and a product page renders Product schema
+        .reduce((acc, current) => {
+          // remove seo properties with falsy values
+          Object.keys(current).forEach(
+            (key) => !current[key] && delete current[key],
+          );
+
+          const {jsonLd} = current;
+
+          if (!jsonLd) {
+            return {...acc, ...current};
+          }
+
+          // concatenate jsonLds if present
+          if (!acc?.jsonLd) {
```

Comment:
After binging on google search console docs last night it seems we need to also support multiple schemas per jsonLd tag. So, jsonLd can be an `object` (single schema) or an array.